yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -976,6 +976,11 @@ private[spark] class Client(
             logError(s"Application $appId not found.")
             return (YarnApplicationState.KILLED, FinalApplicationStatus.KILLED)
           case NonFatal(e) =>
+            if (e.isInstanceOf[InterruptedException]
```

Comment:
Ah sure, you can almost combine the two conditions with `|` in Scala but not quite in this case, but you can at least do ...
```
case e: Exception if e.isInstanceOf[InterruptedException] ||e.getCause.isInstanceOf[InterruptedException] =>
```